### DISCUSSION TITLE
There is no noticeable optimization.

### DISCUSSION THREAD

**Discussion by lwq-star** [optuna/optuna#6061] [Q&A]
### Expected behavior

Why is the optimization result not significant when using the optuna library for hyperparameter optimization of the XGBoost regression model?
This is my hyperparameter optimization code.
```python
def objective(trial):
    params = {
    'n_estimators': trial.suggest_int('n_estimators', 500, 2000),
    'learning_rate': trial.suggest_float('learning_rate', 0.0001, 0.1, log=True),
    'max_depth': trial.suggest_int('max_depth', 10, 28),
    'min_child_weight': trial.suggest_int('min_child_weight', 1, 10),
    'gamma': trial.suggest_float('gamma', 1e-8, 1.0, log=True),
    'subsample': trial.suggest_float('subsample', 0.5, 1.0),
    'colsample_bytree': trial.suggest_float('colsample_bytree', 0.3, 1.0),
    'reg_alpha': trial.suggest_float('reg_alpha', 1e-8, 1.0, log=True),
    'reg_lambda': trial.suggest_float('reg_lambda', 1e-8, 1.0, log=True),
    'random_state': 42
    }
    
    model = XGBRegressor( **params)
    model.fit(x_train, y_train)
    y_pred = model.predict(x_test)
    score = mean_squared_error(y_test, y_pred)
    return score

sampler = optuna.samplers.NSGAIISampler()
pruner = optuna.pruners.HyperbandPruner()
study = optuna.create_study(direction='minimize', sampler=sampler, pruner=pruner)
study.optimize(objective, n_trials=300, show_progress_bar=True, n_jobs = -1)
```
Using `optuna.visualization.matplotlib.plot_optimization_history(study)` can display the optimization history. Despite 300 iterations, the results did not show significant improvement.

![Image](https://github.com/user-attachments/assets/bb5be5ec-9352-4f43-af0e-fcaa263f5913)

![Image](https://github.com/user-attachments/assets/387d117b-6151-4037-9bd1-20b150843daf)

### Environment

- Optuna version:4.2.1
- Python version:3.12.8
- Xgboost:3.0.0
- OS:
- (Optional) Other libraries and their versions:

### Error messages, stack traces, or logs

```shell
no
```

### Steps to reproduce

1.
2.
3.
```python
# python code
```

**Comment by nabenabe0928** [comment]
@lwq-star 
I converted your issue to discussion as your post seems like a question.
Based on my past experience, I believe your problem is rather the performance saturation rather than an optimization issue, meaning that XGBoost cannot be better than the current best trial and Optuna already obtained such a trial in a short amount of time.

**Comment by 1kastner** [comment]
@lwq-star moreover, the later peaks just indicate that TPE sometimes tries to jump out of a local valley / from a local hill and try out something else, adding a bit of exploration to the exploitation. That's the actual advantage of this algorithm over, let's say, a simple hill climbing kind of optimization.
